yarn/src/main/scala/org/apache/spark/deploy/yarn/ClientArguments.scala

```diff
@@ -54,7 +54,8 @@ private[spark] class ClientArguments(args: Array[String], sparkConf: SparkConf)
   private val driverCoresKey = "spark.driver.cores"
   private val amCoresKey = "spark.yarn.am.cores"
   private val isDynamicAllocationEnabled =
-    sparkConf.getBoolean("spark.dynamicAllocation.enabled", false)
+    sparkConf.getBoolean("spark.dynamicAllocation.enabled", false) &&
+      (sparkConf.getInt("spark.executor.instances", 0) == 0)
```

Comment:
There's code later down with comment "Dynamic allocation is not compatible with this option" that should be removed I think. It's superfluous